substrate-node/pallets/pallet-smart-contract/src/storage_state/v9.rs

```diff
@@ -0,0 +1,978 @@
+            types::ContractData::NodeContract(node_contract) => clean_node_contract::<T>(
+                node_contract.node_id,
+                contract_id,
+                node_contract.deployment_hash,
+                &mut r,
+                &mut w,
+            ),
+            types::ContractData::NameContract(_) => clean_name_contract::<T>(),
+            types::ContractData::RentContract(_) => clean_rent_contract::<T>(),
+        }
+
+        // ContractLock
+        if !ContractLock::<T>::contains_key(contract_id) {
+            let now = <timestamp::Pallet<T>>::get().saturated_into::<u64>() / 1000;
+            r.saturating_inc();
+            let mut contract_lock = types::ContractLock::default();
+            contract_lock.lock_updated = now;
+            ContractLock::<T>::insert(contract_id, contract_lock);
+            w.saturating_inc();
+        }
+        r.saturating_inc();
+    }
+
+    debug!(
+        "✨  Smart Contract pallet {:?} cleaning Contracts storage map END",
+        PalletVersion::<T>::get()
+    );
+
+    T::DbWeight::get().reads_writes(r.saturating_add(2), w)
+}
+
+fn clean_node_contract<T: Config>(node_id: u32, contract_id: u64, deployment_hash: HexHash, r: &mut u64, w: &mut u64) {
+    if deployment_hash == HexHash::default() {
+        Contracts::<T>::remove(contract_id);
+        (*w).saturating_inc();
+    }
+
+    // ActiveNodeContracts
+    let mut active_node_contracts = ActiveNodeContracts::<T>::get(node_id);
+    (*r).saturating_inc();
+    if !active_node_contracts.contains(&contract_id) {
+        active_node_contracts.push(contract_id);
+        ActiveNodeContracts::<T>::insert(node_id, active_node_contracts);
+        (*w).saturating_inc();
+    }
+
+    // ContractIDByNodeIDAndHash
+    // Nothing to do here
+    // Storage re-built from zero in clean_contract_id_by_node_id_and_hash()
+    // if !ContractIDByNodeIDAndHash::<T>::contains_key(node_id, deployment_hash) {
+    //     ContractIDByNodeIDAndHash::<T>::insert(node_id, deployment_hash, contract_id);
+    //     *w.saturating_inc();
+    // }
+    // *r.saturating_inc();
+}
+
+fn clean_name_contract<T: Config>() {
+    // Nothing to do here
+}
+
+fn clean_rent_contract<T: Config>() {
+    // Nothing to do here
+}
+
+// ContractsToBillAt
+pub fn clean_contracts_to_bill_at<T: Config>() -> frame_support::weights::Weight {
+    debug!(
+        "🧼  Smart Contract pallet {:?} cleaning ContractsToBillAt storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    let mut r = 0u64;
+    let mut w = 0u64;
+
+    let contracts_to_bill_at = ContractsToBillAt::<T>::iter().collect::<Vec<_>>();
+    r.saturating_accrue(contracts_to_bill_at.len() as u64);
+    let mut contract_id_count = vec![0; (ContractID::<T>::get() + 1) as usize];
+    r.saturating_inc();
+
+    for (_index, contract_ids) in contracts_to_bill_at.clone() {
+        for contract_id in contract_ids {
+            contract_id_count[contract_id as usize].saturating_inc();
```

Comment:
Can't you create the new_contract_ids here? In one loop I mean.

Reply:
Indeed, it can be done

Reply:
Makes sense to do it then to speed up the migration code